internal/manifest/kubernetes.go

```diff
@@ -0,0 +1,268 @@
+		return []string{}, nil
+	}
+
+	if typeMeta.Kind == "Prometheus" {
+		prometheusImages, err := getPrometheusImages(yamlFile)
+		if err != nil {
+			return nil, fmt.Errorf("get prometheus images: %w", err)
+		}
+
+		return prometheusImages, nil
+	}
+
+	if typeMeta.Kind == "Alertmanager" {
+		alertmanagerImages, err := getAlertmanagerImages(yamlFile)
+		if err != nil {
+			return nil, fmt.Errorf("get alertmanager images: %w", err)
+		}
+
+		return alertmanagerImages, nil
+	}
+
+	type BaseSpec struct {
+		Template corev1.PodTemplateSpec `json:"template" protobuf:"bytes,3,opt,name=template"`
+	}
+
+	type BaseType struct {
+		Spec BaseSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	}
+
+	var contents BaseType
+	if err := kubeyaml.Unmarshal(yamlFile, &contents); err != nil {
+		return []string{}, nil
+	}
+
+	var images []string
+	images = append(images, getImagesFromContainers(contents.Spec.Template.Spec.InitContainers)...)
+	images = append(images, getImagesFromContainers(contents.Spec.Template.Spec.Containers)...)
+
+	return images, nil
+}
+
+func getPrometheusImages(yamlFile []byte) ([]string, error) {
+	var prometheus promv1.Prometheus
+	if err := kubeyaml.Unmarshal(yamlFile, &prometheus); err != nil {
+		return nil, fmt.Errorf("unmarshal prometheus: %w", err)
+	}
+
+	var prometheusImage string
+	if prometheus.Spec.BaseImage != "" {
+		prometheusImage = prometheus.Spec.BaseImage + ":" + prometheus.Spec.Version
+	} else {
+		prometheusImage = *prometheus.Spec.Image
+	}
+
+	var images []string
+	images = append(images, getImagesFromContainers(prometheus.Spec.Containers)...)
+	images = append(images, getImagesFromContainers(prometheus.Spec.InitContainers)...)
+	images = append(images, prometheusImage)
+
+	return images, nil
+}
+
+func getAlertmanagerImages(yamlFile []byte) ([]string, error) {
+	var alertmanager promv1.Alertmanager
+	if err := kubeyaml.Unmarshal(yamlFile, &alertmanager); err != nil {
+		return nil, fmt.Errorf("unmarshal alertmanager: %w", err)
+	}
+
+	var alertmanagerImage string
+	if alertmanager.Spec.BaseImage != "" {
+		alertmanagerImage = alertmanager.Spec.BaseImage + ":" + alertmanager.Spec.Version
+	} else {
+		alertmanagerImage = *alertmanager.Spec.Image
+	}
+
+	var images []string
+	images = append(images, getImagesFromContainers(alertmanager.Spec.Containers)...)
+	images = append(images, getImagesFromContainers(alertmanager.Spec.InitContainers)...)
+	images = append(images, alertmanagerImage)
+
+	return images, nil
+}
+
+func getImagesFromContainers(containers []corev1.Container) []string {
+	var images []string
+	for _, container := range containers {
+		images = append(images, container.Image)
+
+		for _, arg := range container.Args {
+			if !strings.Contains(arg, ":") || strings.Contains(arg, "=:") {
+				continue
+			}
+
+			argTokens := strings.Split(arg, "=")
+			images = append(images, argTokens[1])
+		}
+	}
+
+	return images
+}
+
+func contains(images []string, image string) bool {
```

Comment:
Pretty sure there's a similar fucnction, cant locate it now'but it had a more specific name, contains feels too broad.

Reply:
Generics.